thermal-control-design.md

```diff
@@ -0,0 +1,272 @@
+
+These devices shall be included to the temperature monitor list but not limited to: CPU core, CPU pack, ASIC, PSU, Optical Modules, etc.
+
+TEMPERATURE_INFO Table key object_name convention can be "device_name + index"  or device_name if there is no index, like "cpu_core_0", "asic", "psu_2".  Appendix 1 listed all the thermal sensors that supported on Mellanox platform.
+    
+### 2.2 Fan device monitoring 
+
+In most case fan is the device to cool down the switch when the temperature is rising. Thus to make sure fan running at a proper speed is the key for thermal control.
+
+Fan target speed and speed tolerance was defined, by examining them we can know whether the fan reached at the desired speed.
+
+same as the temperature info, a [table for fan](https://github.com/Azure/SONiC/blob/master/doc/pmon/pmon-enhancement-design.md#153-fan-table) also defined as below:
+
+	; Defines information for a fan
+	key                     = FAN_INFO|fan_name              ; information for the fan
+	; field                 = value
+	presence                = BOOLEAN                        ; presence of the fan
+	model                   = STRING                         ; model name of the fan
+	serial                  = STRING                         ; serial number of the fan
+	status                  = BOOLEAN                        ; status of the fan
+	change_event            = STRING                         ; change event of the fan
+	direction               = STRING                         ; direction of the fan 
+	speed                   = INT                            ; fan speed
+	speed_tolerance         = INT                            ; fan speed tolerance
+	speed_target            = INT                            ; fan target speed
+	led_status              = STRING                         ; fan led status
+	timestamp               = STRING                         ; timestamp for the fan info fetched
+
+### 2.3 Syslog for thermal control
+
+If there was warning raised or warning cleared, log shall be generated:
+
+	High temperature warning: PSU 1 current temperature 85C, high threshold 80C！
+	High temperature warning cleared, PSU1 temperature restore to 75C, high threshold 80C
+
+If fan broken or become up present, log shall be generated：
+
+	Fan removed warning: Fan 1 was removed from the system, potential overheat hazard!
+	Fan removed warning cleared: Fan 1 was inserted.
+
+## 3. Thermal control management
+
+Adjust cooling device according to the current temperature can be very vendor specific and some vendors already have their own implementation. In below Appendix chapter describes a Mellanox implementation. But handle the cooling device according to some predefined policies can be generic, this is part of what Thermal control management will do.
+
+This cooling device control function can be disabled if the vendor have their own implementation in the kernel or somewhere else.
```

Comment:
By default, thermal control may be disabled. Platforms (typically lacking BMC) may enable this as necessary.

Reply:
the vendor can disable it in the configuration file so it will be disabled on start, or if the vendor doesn't provide a configuration file, by default it will also be disabled.

Reply:
Thanks!